### PR TITLE
Add MkDocs for documentation

### PR DESCRIPTION
**Pull request by ShaishavGandhi**
Inspired by [OkHttp](https://github.com/square/okhttp)!

* Moves all the Wiki articles to markdown files.
* Created a script to deploy website after releases.

Currently, I've used the Uber Open Source logo and used white color (since that goes with the OSS logo). Open to other colors. 

Sample on my fork: https://shaishavgandhi.github.io/AutoDispose/

### CODE REVIEW

File: docs/css/app.css [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@font-face {
+    font-family: uber-move;
+    src: url("https://d1a3f4spazzrp4.cloudfront.net/dotcom-assets/fonts/UberMove-Medium.woff2") format("woff2");
```
Comment (ShaishavGandhi) [inline docs/css/app.css]: Copied this off the Uber website so not sure if this is the original url.

**Comment by ZacSweers** [comment]
Can you do a writeup of how to maintain and publish this? Can be similar to `RELEASING.md`? Can do as a followup. Will need to update `RELEASING.md` and also possibly adjust osstrich

**Comment by ZacSweers** [comment]
@tyvsmith can you and Brian maybe take a pass at the design/colors question above? I imagine Uber OSS group would like to have some say. Would be good to get an official Uber font link to use as well

**Comment by ShaishavGandhi** [comment]
> Can you do a writeup of how to maintain and publish this? Can be similar to RELEASING.md? Can do as a followup. Will need to update RELEASING.md and also possibly adjust osstrich

Yup, it should be as simple as executing the `deploy_website.sh` script after releasing. Can document that in `RELEASING.md`. 

Have added @tyvsmith as a reviewer as well.

**Comment by ZacSweers** [comment]
In that case, let's get this in but not deploy until the above stuff is handled

**Comment by ShaishavGandhi** [comment]
Sounds good!

**Comment by tyvsmith** [comment]
I'm fine with this as is.

Regarding font link, maybe @briankhsieh can comment on where he got the URL for the main oss site?

Regarding Uber OSS logo, long term, I think autodispose might be a good candidate for it's own logo.

**Comment by briankhsieh** [comment]
@ZacSweers and @tyvsmith can you send me an email. I'll have the designer look into it and provide feedback & needed assets.

**Pull request merged**